### PR TITLE
Adding support for ResolverEndpoint

### PR DESCRIPTION
**Pull request by nishant221**
Issue #, if available:

Description of changes:
Adding support for adding/deleting/updating Route53 ResolverEndpoint in [route53resolver-controller](https://github.com/aws-controllers-k8s/route53resolver-controller) .

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by a-hilaly — COMMENTED** [review]
Can you add some basic e2e tests @nishant221 ?

File: test/e2e/tests/test_resolver_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["RESOLVER_NAME"] = resolver_endpoint
+    replacements["DIRECTION"] = "OUTBOUND"
+    replacements["SUBNET_1"] = get_bootstrap_resources().ResolverEndpointVPC.private_subnets.subnet_ids[0]
+    replacements["SUBNET_2"] = get_bootstrap_resources().ResolverEndpointVPC.private_subnets.subnet_ids[1]
+    replacements["SECURITY_GROUP"] = security_group_id
+
+
+    resource_data = load_route53resolver_resource(
+        "resolver-endpoint",
```
[the remark this answers is not in the document]
Comment (nishant221) [reply test/e2e/tests/test_resolver_endpoint.py]: Changed.

File: test/e2e/tests/test_resolver_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Route53 HostedZone resource
```
[the remark this answers is not in the document]
Comment (nishant221) [reply test/e2e/tests/test_resolver_endpoint.py]: Changed.

File: apis/v1alpha1/resolver_endpoint.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+// (https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_UpdateResolverEndpoint.html)
+// request, a complex type that contains settings for an existing inbound or
+// outbound Resolver endpoint.
+type ResolverEndpointSpec struct {
+
+	// A unique string that identifies the request and that allows failed requests
+	// to be retried without the risk of running the operation twice. CreatorRequestId
+	// can be any unique string, for example, a date/time stamp.
+	// +kubebuilder:validation:Required
+	CreatorRequestID *string `json:"creatorRequestID"`
```
Comment (a-hilaly) [inline apis/v1alpha1/resolver_endpoint.go:38]: Need to ignore this field from the spec - it's marked as required (so the api-server rejects your testing resource) while it's not (you're setting this with a custom ID, in hooks.go)

Comment (nishant221) [reply apis/v1alpha1/resolver_endpoint.go]: Done.

**Comment by nishant221** [comment]
@a-hilaly e2e test is now passing. Kindly have a look.
Thanks

File: pkg/resource/resolver_endpoint/sdk.go [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+	if r.ko.Spec.Name != nil {
+		res.SetName(*r.ko.Spec.Name)
+	}
+	if r.ko.Status.ID != nil {
+		res.SetResolverEndpointId(*r.ko.Status.ID)
+	}
+	if r.ko.Spec.ResolverEndpointType != nil {
+		res.SetResolverEndpointType(*r.ko.Spec.ResolverEndpointType)
+	}
+
```
Comment (a-hilaly) [inline pkg/resource/resolver_endpoint/sdk.go:493]: To complete full support for updates, we ideally want to inject new IPAddresses here.. but it's fine if we do that in a future PR.

**Review by a-hilaly — APPROVED** [review]
Thank you @nishant221! I have comment below, but that can be addressed in a future PR.
/approve
/lgtm

**Comment by a-hilaly** [comment]
Also @nishant221 feel free to open a release PR :)

File: apis/v1alpha1/generator.yaml [hunk 5]
```diff
@@ -3,7 +3,47 @@ ignore:
...
+          input_fields:
+            ResolverEndpointId: Id
+        DeleteResolverEndpoint:
+          input_fields:
+            ResolverEndpointId: Id
+    hooks:
+      sdk_create_post_build_request:
+        template_path: hooks/resolver_endpoint/sdk_create_post_build_request.go.tpl
+    tags:
+      ignore: true
```
Comment (rdtiu) [inline apis/v1alpha1/generator.yaml:49]: new line here is missing. please check your IDE settings.

File: cmd/controller/main.go [hunk 6]
```diff
@@ -18,6 +18,7 @@ package main
 import (
 	"os"
 
+	ec2apitypes "github.com/aws-controllers-k8s/ec2-controller/apis/v1alpha1"
```
Comment (rdtiu) [inline cmd/controller/main.go:21]: some sort of alphabetical order will be more than welcomed :)

File: config/rbac/cluster-role-controller.yaml [hunk 7]
```diff
@@ -31,6 +31,54 @@ rules:
   - list
   - patch
   - watch
+- apiGroups:
```
Comment (rdtiu) [inline config/rbac/cluster-role-controller.yaml:34]: ethos-core labels are missing here.

**Pull request merged**